### PR TITLE
Prep release

### PR DESCRIPTION
**Pull request by Peter9192**
* Add an automated workflow to publish the class web app on GH pages on every release
* Bump version of class package and class web app to 0.0.1
* Fix build by updating the about page
* Update README

Followed examples from [here](https://usagi.io/articles/2024-04-28-deploying-a-solid-start-app-to-github-pages/) and [here](https://dev.to/lexlohr/deploy-a-solid-start-app-on-github-pages-2l2l)

### CODE REVIEW

**Comment by Peter9192** [comment]
Deploy failed due to environment protection rule. Temporarly adding this branch for testing:

image

**Comment by Peter9192** [comment]
Now it worked! Just wonder if we should also add a rule for an allowed "tag" pattern if we want to publish on release.

**Comment by Peter9192** [comment]
@sverhoeven I thought it'd be nice if the GH pages branch is in sync with the latest release, so I've added a GH action workflow that triggers on release and did some other cleaning/release prep. After merging this, I want to make an initial release. Note that I've already tested the workflow by temporarily triggering it for this branch. 

Can you check if there's anything else I should do before we can release this?

**Comment by sverhoeven** [comment]
`@repo` should be replace with `@classmodel` to publish to npmjs

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  contents: read
+  pages: write
+  id-token: write
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (sverhoeven) [inline .github/workflows/deploy.yml:17]: You are not using the latest actions.

```suggestion
      - uses: actions/checkout@v4
```

File: apps/class-solid/src/routes/about.tsx [hunk 2]
```diff
@@ -1,17 +1,14 @@
 import { A } from "@solidjs/router";
-import Counter from "~/components/Counter";
```
Comment (sverhoeven) [inline apps/class-solid/src/routes/about.tsx:2]: Ah I made similar change in https://github.com/classmodel/class-web/pull/8/commits/a4549ac03cc7f52bc6424351673f6334ab24e405, we will need to resolve that conflict later.

File: apps/class-solid/src/routes/index.tsx [hunk 3]
```diff
@@ -12,6 +12,7 @@ import { analyses } from "~/lib/store";
 import { addAnalysis, AnalysisCard } from "~/components/Analysis";
 
 export default function Home() {
+  console.log(import.meta.env.SERVER_BASE_URL);
```
Comment (sverhoeven) [inline apps/class-solid/src/routes/index.tsx:15]: Please remove this debug statement.

```suggestion
```

File: README.md [hunk 4]
```diff
@@ -1,33 +1,74 @@
-# `Turborepo` Vite starter
+# CLASS-web
 
-This is an official starter Turborepo.
+This is an implementation of CLASS that runs entirely in the browser.
```
Comment (sverhoeven) [inline README.md:3]: Would be nice to spell out CLASS, so a random visitor can get quicker insight what this repo is about.

```suggestion
This is an implementation of  the **C**hemistry **L**and-surface **A**tmosphere **S**oil **S**lab (CLASS) model that runs entirely in the browser.
```

File: README.md [hunk 5]
```diff
@@ -1,33 +1,74 @@
...
-This Turborepo includes the following packages and apps:
+## Local build
+
+To run a local development version:
+
+```sh
+git clone git@github.com:classmodel/class-web.git
+cd class-web/apps/class-solid
+pnpm install
+pnpm run dev
```
Comment (sverhoeven) [inline README.md:38]: Can be simplified to

```suggestion
cd class-web
pnpm install
pnpm dev
```

This also makes sure that if a package needs building, that that is done before the app is ready. Thanks to turborepo

File: README.md [hunk 6]
```diff
@@ -1,33 +1,74 @@
...
-- `@repo/typescript-config`: `tsconfig.json`s used throughout the monorepo
+The CLASS package is written in typescript. It uses [zod](https://zod.dev/) for
+the configuration and runtime validation. Zod can be converted to/from
+[JSONSchema](https://json-schema.org/), which is ideal for sharing the
+configuration between web-app, library code, and perhaps other implementations
+of CLASS as well.
 
-Each package and app is 100% [TypeScript](https://www.typescriptlang.org/).
+The web application is build with [solid.js](https://docs.solidjs.com/). Solid
+has a relatively simple model for building reactive web applications. With its
```
Comment (sverhoeven) [inline README.md:50]: They call them selves a framework at https://docs.solidjs.com/

```suggestion
is a relatively simple framework for building reactive web applications. With its
```

We are already using model for class, let us not used it to describe a framework/library as well.

**Review by sverhoeven — APPROVED** [review]
The deployment at https://github.com/classmodel/class-web/actions/runs/10038150263 looks good.

Some minor nit picks.

**Comment by Peter9192** [comment]
Thanks for the suggestions! Merging this before reviewing #9

**Pull request merged**